plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -0,0 +1,577 @@
+{-# LANGUAGE DeriveAnyClass    #-}
+{-# LANGUAGE DerivingVia       #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+{-# LANGUAGE TemplateHaskell   #-}
+{-# OPTIONS_GHC -Wno-simplifiable-class-constraints #-}
+{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}
+{-# OPTIONS_GHC -fno-specialise #-}
+{-# OPTIONS_GHC -fno-strictness #-}
+
+module PlutusLedgerApi.V3.Contexts where
+
+import GHC.Generics (Generic)
+import Prettyprinter (nest, vsep, (<+>))
+import Prettyprinter.Extras
+
+import PlutusLedgerApi.V2 qualified as V2
+import PlutusTx qualified
+import PlutusTx.AssocMap hiding (filter, mapMaybe)
+import PlutusTx.Prelude qualified as PlutusTx
+
+import Prelude qualified as Haskell
+
+newtype ColdCommitteeHash = ColdCommitteeHash V2.Credential
```

Comment:
Yeah, it is good practice :slightly_smiling_face: 

Before there was only really only Payment and Staking credentials, now we have 3 more kinds. It'll be good to distinguish between them. For consistency sake it might be good to add `PaymentCredential` too. Just a suggestion.


Also, suggestion for my naming mishap: Credential
```suggestion
newtype ColdCommitteeCredential = ColdCommitteeCredential V2.Credential
```